modules/shadertools/src/lib/filters/prop-types.ts

```diff
@@ -1,20 +1,31 @@
+// luma.gl, MIT license
+
+/** 
+ * For use by shader module and shader pass writers to describe the types of the 
+ * properties they expose (properties ultimately map to uniforms).
+ */
 export type PropType = {
   type?: string;
   max?: number;
   min?: number;
   value?: any;
 } | number;
 
-export type PropDef = {
+/**
+ * Internal property validators generated by processing the prop types , 
+ * The `validate()` method can be used to validate the type of properties passed in to
+ * shader module or shader pass
+ */
+export type PropValidator = {
   type: string;
   value: any;
   max?: number;
   min?: number;
   private?: boolean;
-  validate?(value: any, propDef: PropDef): boolean;
+  validate?(value: any, propDef: PropValidator): boolean;
```

Comment:
```suggestion
  validate?(value: any, propType: PropType): boolean;
```

Reply:
The original propType isn't saved so it may not be available.

I moved the validation code into the prop-types file which I think makes things a lot clearer.

I think that is sufficient amount of improvements for this round